ImprovedAnvil/lib/spells/03_patches_vanilla.tph

```diff
@@ -451,25 +1194,14 @@ COPY_EXISTING ~writhing.pro~ ~override/writhing.pro~
 
 // patch in removal of shaman's innate aura cleansing //
 
-COPY_EXISTING_REGEXP GLOB ~SPPR[1-7][0-5][0-9]+\.spl~ ~override~
-                          ~SPDR202.spl~ ~override~
-                          ~SPDR302.spl~ ~override~
-                          ~SPCL940.spl~ ~override~
-                          ~SPCL941.spl~ ~override~
-                          ~SPCLI12.spl~ ~override~
-                          ~SPCLI15.spl~ ~override~
-                          ~SPCLI16.spl~ ~override~
-  READ_BYTE 0x21 exc_flags
-
-  PATCH_IF ((exc_flags BAND BIT7) = 0) BEGIN
-    LPF ADD_SPELL_EFFECT
-      INT_VAR
-        opcode = 321
-        target = 1
-        insert_point = 0
-      STR_VAR
-        resource = ~SPPRI25~
-    END
+COPY_EXISTING_REGEXP GLOB ~.+\.spl~ ~override~
```

Comment:
why is this modified?

Reply:
Take a look at Purify Aura and Cause X Wounds spells. Both of them use SPPRI25. And since you can make almost any combination of classes and kits with Clerics or Druids (who get Cause X Wounds spells), it's better to simply patch all spells, so there is no chance to miss some spell/ability, that could be used by players.

Reply:
This is a very unrestrictive approach that will add an unnecessary condition to ALL spells in the game, I'm against it. It creates potential for undesired interactions and extra clutter that needs to be kept in mind when working with any spl file in the game whatsoever. This best be kept to the actual spells the character can cast from a spell book.

Reply:
It might be very hard to ensure that all player-usable spells and no spells unavailable to players are included.

Reply:
Well, if you take a look at the original version, you'll see it wasn't so hard to discover all druidic spells. It is basically SPPRXXX (which are standard naming scheme for priest spells that could go in a spellbook) and a few extra HLAs with custom names. The only trick here was to read the druid/ranger/shaman exclusion flag and make sure it is not set.

Reply:
Now we will have to remember about non-standard spells and abilities, like ohtyr1, ohtyr2, which for some reason are special abilities of Priest of Tyr, same thing with Tempus. It's a good thing that new IA spells still adhere to the original naming convention with SPPR/SPWI/SPIN/SPINI/SPCL.

Reply:
Okay, now only those abilities that can be used by players get this dispell of Aura Cleanse. I tried to exclude also abilities of kits that can't dual class and can't get Aura Cleanse in any form.

Reply:
Yeah, it's definitely better to load the spell list from a custom 2da at some point, but not critical at the moment

Reply:
could you put please a // @TODO comment about this into the code and resolve this thread for now

Reply:
ok, so for now I will resolve this conversation.